creator/ingest_runs/models.py

```diff
@@ -73,6 +82,35 @@ class Meta:
         help_text="The user who submitted this ingest run",
     )
 
+    @transition(field=state, source="waiting", target="running")
+    def start(self):
+        """
+        Begin running the ingest process.
+        """
+        return
+
+    @transition(field=state, source="running", target="complete")
+    def complete(self):
+        """
+        Finish running the ingest process without error.
+        """
+        return
+
+    @transition(field=state, source="*", target="canceled")
+    def canceled(self):
+        """
+        Cancel the ingest run on request from the user or on account of a new
+        ingest run being started for the same input parameters.
+        """
+        return
+
+    @transition(field=state, source="*", target="failed")
```

Comment:
Also not sure about this but I think you can only get into failed state from running or canceled. The ingest run could be running and then unexpectedly fail or the ingest run could be canceled by the user causing it to go into failed state.